app/controllers/sac_cas/people_controller.rb

```diff
@@ -20,8 +22,62 @@ def list_filter_args
     Person::Filter::NeuanmeldungenList.new(group, current_user).filter_params
   end
 
+  def update(options = {}, &block)
+    assign_attributes
+    if validate_new_manageds
+      updated = with_callbacks(:update, :save) { save_entry }
+      respond_with(entry, options.reverse_merge(success: updated, location: return_path), &block)
+    else
+      render :edit
+    end
+  end
+
+  def assign_roles_for_newly_created_managed_people
+    return unless newly_created_managed_people.any?
+    
+    Role.transaction do
+      cloneable_mitglied_roles.each do |role|
+        newly_created_managed_people.each do |p|
+          Role.create!(person: p,
+                       group: role.group,
+                       type: role.type,
+                       beitragskategorie: role.beitragskategorie,
+                       created_at: Time.zone.now,
+                       delete_on: role.delete_on)
+        end
+      end
+    end
+  end
+
+  def validate_new_manageds
+    validate_entry_age! && validate_new_manageds_age!
+  end
+
+  def cloneable_mitglied_roles
+    entry.roles.where(type: SacCas::MITGLIED_ROLES, beitragskategorie: 'familie')
+  end
+
+  def newly_created_managed_people
+    @newly_created_managed_people ||= new_manageds.select(&:new_record?)
+  end
+
   private
 
+  def validate_entry_age!
+    return true unless entry.years <= 22
```

Comment:
maybe reference to this? https://github.com/hitobito/hitobito_sac_cas/blob/master/app/domain/sac_cas/beitragskategorie/calculator.rb#L12

assert_sac_adult instead of validate_entry_age! ?